packs/tfe_fdo_nomad/templates/tfe.nomad.tpl

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+job [[ .tfe_fdo_nomad.job_name | quote ]] {
+  datacenters = [[ .tfe_fdo_nomad.datacenters  | toStringList ]]
+  namespace   = [[ .tfe_fdo_nomad.tfe_namespace | quote ]]
+  type        = "service"
+
+  group "tfe-group" {
+    count = [[ .tfe_fdo_nomad.tfe_group_count ]]
+    
+    spread {
+      attribute = "${node.unique.id}"
+    }
+
+    restart {
+      attempts = 3
+      delay    = "60s"
+      interval = "10m"
+      mode     = "fail"
+    }
+
+    update {
+      max_parallel      = 1
+      min_healthy_time  = "30s"
+      healthy_deadline  = "12m"
+      progress_deadline = "15m"
+      health_check      = "checks"
+    }
+
+
+    network {
+      port "tfe" {
+        static = [[ .tfe_fdo_nomad.tfe_port ]]
+      }
+      port "http" {
+        static = [[ .tfe_fdo_nomad.tfe_http_port ]]
+      }
+      port "vault" {
+        static = [[ .tfe_fdo_nomad.tfe_vault_cluster_port ]]
+      }
+    }
+
+    service {
+      name     = [[ .tfe_fdo_nomad.tfe_service_name | quote ]]
+      port     = "tfe"
+      provider = [[ .tfe_fdo_nomad.tfe_service_discovery_provider | quote ]]
+
+      check {
+        type     = "http"
+        port     = "http"
+        path     = "/_health_check"
+        interval = [[ .tfe_fdo_nomad.health_check_interval | quote ]]
+        timeout  = [[ .tfe_fdo_nomad.health_check_timeout | quote ]]
+      }
```

Comment:
Given how long the `healthy_deadline` is, you may want to look into splitting the health check into a readiness and liveness check. See the [`check.on_update`](https://developer.hashicorp.com/nomad/docs/job-specification/check#on_update) block, and maybe also the [`check_restart`](https://developer.hashicorp.com/nomad/docs/job-specification/check_restart) block for the liveness check.

Reply:
I don't understand this point, I have only 1 health API which gives an aggregated health check for all the services inside the container.

Is there a way to break this single API into liveness and readiness checks?

Reply:
No, you'd need separate health APIs to do separate liveness and readiness checks.

Reply:
doubted so, then we will have to keep it as it is or do you suggest something else?

Reply:
Seems fine. I'd want to fix this upstream in the application if I had to maintain it. But if it's not there, the Pack can't use it. :grin: